### ISSUE TITLE
`deposit` accepts more than the minimum input size

### ISSUE THREAD

**Issue by ralexstokes** [ethereum/deposit_contract#20]
`deposit` currently accepts an input of size `512`.

The current required size of the deposit input is a BLS pubkey (48 bytes), a 32 byte hash and a BLS signature (96 bytes). For a total of 176 bytes.

Any reason we accept a deposit input of size greater than this? I'm not sure if the Vyper compiler generates code to check the runtime length against the type (and if we would even want this) but figured I would bring it up. An argument for leaving it as is would be that it is "big enough" (and is a nice power of two) so that if the format of the deposit input changes (e.g. expands) then we don't have to change this contract.

**Comment by NIC619** [comment]
IIRC it is just a "big enough" number(originally it was 2048) but I could be wrong…

I suppose now that we have to [check input size](https://github.com/ethereum/eth2.0-specs/issues/710), it should just be `8+8+48+32+96=192` bytes. I will update the contract accordingly.

**Comment by NIC619** [comment]
Ah or maybe update it together in #21 ? @ralexstokes

**Comment by ralexstokes** [comment]
@NIC619 i'd prefer to keep this issue to a different PR

and it sounds like we want to adjust the input type of `deposit` to `bytes[176]` and also insert an assertion that the final serialized deposit data is of length `192` bytes. (those values may be +/- some bytes given ssz serialization overheads)

**Comment by ralexstokes** [comment]
its also worth holding off on a PR for this until the serialization discussions stabilize a bit

**Comment by ralexstokes** [comment]
looks like the latest on this is: https://github.com/ethereum/eth2.0-specs/issues/710#issuecomment-484759178

tl;dr: we require the (little-endian) serialized parameters as distinct inputs to `deposit`.
